extensions/README.md

```diff
@@ -0,0 +1,152 @@
+build your own custom project template.
+
+https://github.com/liferay/liferay-blade-cli/tree/master/extensions/sample-template
+
+Here are a few things to notice from this sample.
+
+1. This template is structured as a Maven archetype.  Read more about Maven Archetypes and their features and capabilities here: https://maven.apache.org/guides/introduction/introduction-to-archetypes.html
+2. The BundleSymbolic name of your template jar (see here: https://github.com/liferay/liferay-blade-cli/blob/master/extensions/sample-template/bnd.bnd#L3) must have the pattern `*.project.templates.<name>.*`  So lets say you have a custom template called "mywebapp", then the BSN should be `org.myorg.project.templates.mywebapp-1.0.0.jar`
+3. Testing your custom template can be accomplished by following the example here: https://github.com/liferay/liferay-blade-cli/blob/master/extensions/sample-template/src/test/java/com/liferay/project/templates/sample/TemplatesTest.java
+
+Once you have your custom template you can install it by just copying to ${user.home}/.blade/extensions or using the
+`$ blade extension install /path/to/org.myorg.project.templates.mywebapp-1.0.0.jar`
+
+** Give it a try! **
+```
+$ blade create -l
+```
+Now you should see your new custom project template listed!.  If not, contact us on our #blade forum on the Liferay community slack channel.  https://community.liferay.com/it/chat
+
+### Blade Profiles
+
+Lets say that you want to customize the normal blade development workflow.  Normally Liferay developers who use blade will run a series of blade commands that all make sense in the "default" lifeary workspace.
+
+1. blade init
+2. blade server init
+3. blade server start
+4. blade create my-project
+5. blade deploy
+6. blade customCommand
+
+But if this workspace wants to support `docker` or another _containerized_ Liferay environment, all of those commands
+should do something slightly different.  And the case of the last command, it is a *new* command that only makes sense
+inside of this development environment.
+
+So a blade profile allows extensions developers to "override" existing blade commands or add "new" commands.
+So in the workflow above, blade init for a profile "myprofile" would override the default "init" command do something
+before/after the normal init command.  `blade server start` in the case of a profile would override the `ServerStartCommand` in blade and contribute its own way to start the Liferay server (maybe in a container of some sort).
+
+`blade deploy` or `blade watch` would likely mean something completely different in the context of a profile, so that command
+would need to be overridden.
+
+The way to "override" existing commands can be see here in the example: https://github.com/liferay/liferay-blade-cli/tree/master/extensions/sample-profile/src/main/java/com/liferay/extensions/sample/profile
```

Comment:
Perhaps we should also reference / remind the user about the Service Loader file here? `https://github.com/liferay/liferay-blade-cli/blob/master/extensions/sample-profile/src/main/resources/META-INF/services/com.liferay.blade.cli.command.BaseCommand`